src/components/RangeSlider/components/DualThumb/DualThumb.scss

```diff
@@ -86,39 +95,54 @@ $range-thumb-shadow-focus: 0 0 0 rem(1px) color('indigo');
   width: $range-thumb-size;
   height: $range-thumb-size;
   border-radius: 50%;
-  border: border-width() solid color('sky', 'lighter');
+  border: border-width() solid
+    var(--p-action-interactive, color('sky', 'lighter'));
   box-shadow: $range-thumb-shadow;
-  background: linear-gradient(color('sky', 'lighter'), color('sky', 'light'));
+  background: linear-gradient(
+    var(--p-action-interactive, color('sky', 'lighter')),
+    var(--p-action-interactive, color('sky', 'light'))
+  );
   -webkit-tap-highlight-color: transparent;
 
   // stylelint-disable-next-line value-no-vendor-prefix
   cursor: -webkit-grab;
+  transition-property: width, height, margin;
+  transition-duration: duration();
+  transition-timing-function: easing();
 
   &.disabled {
     cursor: not-allowed;
-    border-color: color('sky', 'dark');
+    border-color: var(--p-border-secondary-disabled, color('sky', 'dark'));
+    background: var(--p-border-secondary-disabled, color('white'));
   }
 
   &:hover {
-    background: linear-gradient(color('sky', 'lighter'), color('sky', 'light'));
     box-shadow: $range-thumb-shadow-hover;
   }
 
-  .error & {
-    border-color: color('red');
-    box-shadow: $range-thumb-shadow-error;
+  &:active {
+    margin-left: var(--p-range-slider-thumb-offset);
```

Comment:
This is needed so that the thumb remains "centered" as it grows in size.

Since it is positioned absolutely, and its width/height increase when active, it grows from the top left corner, causing it to be misaligned with the output bubble.

I tried modifying the `left` value in the JSX, but this would require the `useFeatures` hook, which would require functionalizing this component.

Reply:
You cant use `Class.contextType` to access `this.context` in a class component.

- https://reactjs.org/docs/context.html#classcontexttype

Reply:
Which approach do you think is cleaner?

I went with the css because the value of the thumb offset is actually the negative difference of the active thumb size value and base value (`-4px`). I felt that modifying the JSX might cause future "where is that value coming from?"

Reply:
If we decide to go with the css approach, it'll probably be clearer to do that math instead of hardcoding a new variable `rangeSliderThumbOffset`.

Reply:
Aren't we already doing the work in JavaScript for the existing product design language? No sense in doing it in two places. We also try to avoid applying `margin` within our components, unless they are specifically for layout.

Reply:
Yup we are, but it doesn't currently account for the thumb changing in size when active.

Reply:
Scratch all this! Looks like we don't need to reposition the thumb to the output bubble if we scale the thumb as per this suggestion https://github.com/Shopify/polaris-react/pull/2655#discussion_r371540536